### PR TITLE
Fix stock on hand report from locations with special characters

### PR DESCRIPTION
**Pull request by gcapalbo**
fixes http://manage.dimagi.com/default.asp?186625
@emord

### CODE REVIEW

**Comment by czue** [comment]
looks good to me, though wasn't clear how it fixes the linked ticket

**Comment by gcapalbo** [comment]
@czue See my response to Kelly for the details. The error from the title in the ticket was being raised from some old supply functionality that Mike said we can likely remove.  The error that this fixes is the second email I attached in the ticket, which was a different issue but was still preventing Kelly from submitting stock reports.

**Comment by czue** [comment]
aha - missed that correspondence. thanks1

File: corehq/apps/locations/models.py [hunk 1]
```diff
@@ -438,7 +438,8 @@ def __init__(self, *args, **kwargs):
             self.location_type = location_type
 
     def __repr__(self):
-        return "%s (%s)" % (self.name, self.location_type)
+        val = u"%s (%s)" % (self.name, self.location_type)
+        return val.encode('utf-8')
```
Comment (esoergel) [inline corehq/apps/locations/models.py]: @gcapalbo Any tips on where I can find this being called?  I'd like to provide a more helpful dev-facing representation (which includes the type of the object), but I wanna make sure I don't also make that user-facing.

Comment (gcapalbo) [inline corehq/apps/locations/models.py]: @esoergel I'm not sure, I haven't really worked on much supply stuff, but the place where it was being called from to cause the issue that this PR fixed was here: https://github.com/dimagi/commcare-hq/blob/0fcff600792ef1a7239a890f8c7f15617d23e0e7/corehq/apps/commtrack/sms.py#L68

Comment (esoergel) [inline corehq/apps/locations/models.py]: Okay, nice, thanks.  I was worried we had something like `"Your location is {}".format(location)`, where the user would see the `__repr__`

**Pull request merged**